### PR TITLE
feat: add data layer for persisted filter settings

### PR DESCRIPTION
**Pull request by matthiasn**
## Summary by CodeRabbit

* **New Features**
  * Save and pin task filters in the sidebar for quick access
  * Create, rename (trimmed input), update filter criteria, delete, and reorder saved filters
  * Saved filters persist across sessions and preserve order; redundant writes are avoided

* **Tests**
  * Added unit tests covering JSON serialization, controller operations, persistence behavior, and reordering edge cases

### CODE REVIEW

File: lib/features/tasks/state/saved_filters/saved_task_filters_controller.dart [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+      id: _uuid.v4(),
+      name: name,
+      filter: filter,
+    );
+    final current = state.value ?? const <SavedTaskFilter>[];
+    final next = [...current, saved];
+    state = AsyncData(next);
+    await _persistence.save(next);
+    return saved;
+  }
```
[the remark this answers is not in the document]
Comment (matthiasn) [reply lib/features/tasks/state/saved_filters/saved_task_filters_controller.dart:45]: Fixed. All five mutation methods (`create`, `rename`, `updateFilter`, `delete`, `reorder`) now `await future` before reading state, so they wait for the initial load to complete and never overwrite persisted data with an empty list. Added a regression test that calls `create` immediately on the notifier without first awaiting the build future and asserts the prior persisted entry is preserved.
